kbase-extension/static/kbase/js/common/cellComponents/filePathWidget.js

```diff
@@ -663,6 +683,97 @@ define([
             });
         }
 
+        /**
+         * Steps.
+         * 1. Get all values from output parameters, ignore null values
+         * 2. Filter down to those with duplicates
+         * 3. Send those to just the inputs those came from.
+         * 4. Up to widgets to act accordingly
+         * 5. Return the row ids that are duplicates
+         */
+        function updateDuplicateOutputValues() {
+            const outputValueCounts = model.getItem('outputParamIds').reduce((counts, paramId) => {
+                const outputValues = getParameterValuesByRow(paramId);
+                for (const [rowId, value] of Object.entries(outputValues)) {
+                    if (!value) {
+                        return counts;
+                    } else if (!(value in counts)) {
+                        counts[value] = [];
+                    }
+                    counts[value].push({ rowId, paramId });
+                }
+                return counts;
+            }, {});
+            const values = Object.keys(outputValueCounts).reduce((duplicateValues, value) => {
+                if (outputValueCounts[value].length > 1) {
+                    duplicateValues[value] = outputValueCounts[value];
+                }
+                return duplicateValues;
+            }, {});
+            // values should now look like:
+            // {
+            //   'field_value': [{ rowId1, paramId }, { rowId2, paramId }]
+            // }
+            // with only entries where there are at least two row ids
+            // translate to something easier to work with -
+            // {
+            //    rowId: {
```

Comment:
IMO, this is annoyingly complex. But I think there will (eventually) be uploaders that have multiple output objects. For iterating, this seems the easiest.

Reply:
Should I assume that 'row' refers to everything within a section marked with a number? For example, this screenshot shows two rows:
<img width="474" alt="Screenshot 2021-08-17 at 07 01 21" src="https://user-images.githubusercontent.com/556057/129739890-e0892408-0b10-4cd6-9b11-cd509d4fd8fe.png">

Reply:
Yeah, each "row" is one set of file path inputs, like your screenshot.